.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,60 @@
+name: Docker Image
+on:
+  push:
+    tags:
+      - '**'
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+    - uses: hadolint/hadolint-action@54c9adbab1582c2ef04b2016b760714a4bfde3cf  # v3.1.0
+      with:
+        verbose: true
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3  # v3.0.0
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226  # v3.0.0
+      with:
+        install: true
+
+    - name: Extract tags/labels from Git
+      id: docker_meta
+      uses: docker/metadata-action@9dc751fe249ad99385a2583ee0d084c400eee04e  # v5.4.0
+      with:
+        images: jaegertracing/thrift
+        # The 'tags:' section defines how the Docker image will be tagged:
+        #   - pushes to master branch will be published as 'latest'
+        #   - pushes tagged with semver will be published as that version (without 'v')
+        #   - other tags can be used as is
+        # Documentation: https://github.com/docker/metadata-action#tags-input
+        tags: |
+          type=raw,value=latest,enable=${{ github.ref == 'refs/heads/master' }}
+          type=semver,pattern={{version}}
+          type=ref,event=tag
+
+    - name: Login to DockerHub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d  # v3.0.0
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
These were missing so the workflow failed. I just added them. Can you add `on: workflow_dispatch:` to the workflow triggers?